services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/EventController.java

```diff
@@ -0,0 +1,1023 @@
+
+        if (testData) {
+            List<MapMinimumDataEvent> list = new ArrayList<>();
+            return ResponseEntity.ok(list);
+        } else {
+            Query query = mapMinimumDataEventRepo.getQuery(intersectionID, startTime, endTime, latest);
+            return ResponseEntity.ok(mapMinimumDataEventRepo.find(query));
+        }
+    }
+
+    @RequestMapping(value = "/events/map_minimum_data/count", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("@PermissionService.isSuperUser() || (@PermissionService.hasIntersection(#intersectionID) and @PermissionService.hasRole('USER')) ")
+    public ResponseEntity<Long> countMapMinimumDataEvents(
+            @RequestParam(name = "intersection_id", required = false) Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis", required = false) Long startTime,
+            @RequestParam(name = "end_time_utc_millis", required = false) Long endTime,
+            @RequestParam(name = "full_count", required = false, defaultValue = "true") boolean fullCount,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            return ResponseEntity.ok(1L);
+        } else {
+            Query query = mapMinimumDataEventRepo.getQuery(intersectionID, startTime, endTime, false);
+            long count;
+
+            if (fullCount) {
+                count = mapMinimumDataEventRepo.getQueryFullCount(query);
+            } else {
+                count = mapMinimumDataEventRepo.getQueryResultCount(query);
+            }
+
+            log.debug("Found: {} MapMinimumDataEvents", count);
+            return ResponseEntity.ok(count);
+        }
+    }
+
+    @RequestMapping(value = "/events/map_broadcast_rate", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("@PermissionService.isSuperUser() || (@PermissionService.hasIntersection(#intersectionID) and @PermissionService.hasRole('USER')) ")
+    public ResponseEntity<List<MapBroadcastRateEvent>> findMapBroadcastRateEvents(
+            @RequestParam(name = "intersection_id", required = false) Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis", required = false) Long startTime,
+            @RequestParam(name = "end_time_utc_millis", required = false) Long endTime,
+            @RequestParam(name = "latest", required = false, defaultValue = "false") boolean latest,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            List<MapBroadcastRateEvent> list = new ArrayList<>();
+            list.add(MockEventGenerator.getMapBroadcastRateEvent());
+            return ResponseEntity.ok(list);
+        } else {
+            Query query = mapBroadcastRateEventRepo.getQuery(intersectionID, startTime, endTime, latest);
+            return ResponseEntity.ok(mapBroadcastRateEventRepo.find(query));
```

Comment:
issue(blocking): executing query second time for logging purposes

Reply:
Extra query has been removed